src/main/java/com/github/martynagil/drugstoremanagement/service/EmployeeService.java

```diff
@@ -35,9 +35,11 @@ public void addNewEmployee(EmployeeDto employeeDto) {
     public void dismissEmployee(Long employeeId, EmployeeDismissalDto employeeDismissalDto) {
         Employee employee = employeeRepository.findById(employeeId)
                 .orElseThrow(() -> new EntityNotFoundException());
-        if (employee.getDateOfDismissal().equals(null)) {
+        if (employee.getDateOfDismissal() == null) {
```

Comment:
to co tutaj jest to nie jest programowanie obiektowe. Zamiast sterować logiką obiektu z zewnątrz, spytaj obiekt o jego stan. Inaczej: spytaj pracownika czy jest zwolniony